### ISSUE TITLE
VS Code feature request: Inline plots that respect chunk dimension options

### ISSUE THREAD

**Issue by andrewheiss** [quarto-dev/quarto#287]
In RStudio, chunks that generate plots follow the dimensions specified in the chunk options (currently only with in-chunk options, but [eventually hopefully also with YAML-style chunk options too](https://github.com/rstudio/rstudio/issues/12305))

<img width="879" alt="image" src="https://github.com/quarto-dev/quarto/assets/73663/c3c1674f-46b0-406a-a5fc-e88f4b41d294">

&nbsp;

In VS Code, images do not appear inline and instead either show up in a separate panel (if {httpgd} is installed), or in a separate window:

<img width="1332" alt="image" src="https://github.com/quarto-dev/quarto/assets/73663/2029882c-12e2-45fd-90bc-1775419c040e">

&nbsp;

This is also the case when using the visual editor mode:

<img width="1278" alt="image" src="https://github.com/quarto-dev/quarto/assets/73663/a8bfb939-bb67-4607-9651-b762f46bfc00">

&nbsp;

It would be helpful for figuring out figure dimensions for documents if plots could appear using the specified chunk options. But that also might just be a limitation of VS Code and/or radian and/or {httpgd} and/or something else.

**Comment by jjallaire** [comment]
Unfortunately we are sending plots to the R extension so can't directly manipulate the figure size as we can in RStudio. If there is a way to do this that we aren't aware of we are all ears to implement it!

**Comment by Pindar777** [comment]
@andrewheiss could the use of `out-width: X%` als command inside the r-chunk be a solution?

**Comment by andrewheiss** [comment]
> @andrewheiss could the use of `out-width: X%` as command inside the r-chunk be a solution?

Since the plot appears in a separate window, changing the `out-width` doesn't change anything, and if the plot could appear inline like in RStudio,`out-width` would only modify the width, not the height, and wouldn't change the actual dimensions of the image, just how it is displayed.

<img width="951" alt="image" src="https://github.com/quarto-dev/quarto/assets/73663/7577faa8-45c0-4836-bc9c-95c56e830efa">

**Comment by Pindar777** [comment]
@andrewheiss I used the mentioned option to shrink images in docx output since fig.height etc. did not work (as opposed to bookdown).

Your description of the window-output makes perfectly sense. In VSCode I just resize that window, though.

**Comment by MichelNivard** [comment]
> Unfortunately we are sending plots to the R extension so can't directly manipulate the figure size as we can in RStudio. If there is a way to do this that we aren't aware of we are all ears to implement it!

Could you send the R code to usr/bin/Rscript via the terminal instead of to R? Then you could catch the output and render inline? Or does either vscode or the design philosophy of knitr prevent this?

**Comment by venpopov** [comment]
I'm bumping up this issue. I recently switched entirely to vscode for all my R and quatro coding, and the ability to render code output and plots is what I miss the most from Rstudio. Could the visual editor somehow take advantage of the the Notebook API: https://code.visualstudio.com/api/extension-guides/notebook which allows to serialize code chunk outputs, save them to disk and render them?

**Comment by andrewheiss** [comment]
Bumping this again now that Positron is a thing. Because Positron is essentially VS Code, inline images don't appear inline, and they don't take the chunk dimensions

<img width="1317" alt="image" src="https://github.com/quarto-dev/quarto/assets/73663/c006bd11-3a1a-4688-b50b-a7d3fb4543e9">

**Comment by stefanocoretta** [comment]
Yup, I too really miss the inline output feature of RStudio (https://github.com/posit-dev/positron/discussions/4033#discussioncomment-11789729).
